input/init.go

```diff
@@ -26,5 +26,5 @@ func SetLogger(l *logging.Logger) {
 }
 
 type Handler interface {
-	Handle(net.Conn)
+	Handle(io.Reader)
```

Comment:
can you document the packet oriented vs conn oriented stuff? not sure where is the best place, probably in each implementation.  but this method can probably also use a comment describing how this is supposed to work because handling a reader doesn't seem as intuitive as handling a conn.

Reply:
The handler accepting an io.Reader has nothing to do with packet oriented vs conn oriented.
The handling of UDP has changed to be packet oriented instead of conn oriented, but that is just because that is the proper way to process UDP.  UDP is a connectionless protocol.

Although the handlers were being passed a conn, they were only using it as a io.Reader, using it to create a new bufio.Reader.  Golang best practices state that you should only require the functionality that you need.  As the handler only reads, it only requires a reader.

https://dave.cheney.net/2016/08/20/solid-go-design 
 under "Interface Segregation Principle
"